lib/datetime/datetime.ex

```diff
@@ -447,13 +447,17 @@ defimpl Timex.Protocol, for: DateTime do
           err
 
         %DateTime{} = datetime when shift != 0 ->
-          DateTime.add(datetime, shift, :microsecond, Timex.Timezone.Database)
+          datetime
+          |> DateTime.add(shift, :microsecond, Timex.Timezone.Database)
+          |> retain_precision(datetime)
 
         %DateTime{} = datetime ->
           datetime
 
-        {{ty, _, _}, %DateTime{} = orig} when ty in [:gap, :ambiguous] and shift != 0 ->
-          DateTime.add(orig, shift, :microsecond, Timex.Timezone.Database)
+        {{ty, _, _}, %DateTime{} = original} when ty in [:gap, :ambiguous] and shift != 0 ->
+          original
+          |> DateTime.add(shift, :microsecond, Timex.Timezone.Database)
```

Comment:
Maybe it's a good idea to create a function for adding microseconds while keeping the precision? Something like:

```elixir
original |> add_with_precision_preserved(shift)
```